circle.yml

```diff
@@ -0,0 +1,43 @@
+machine:
+  timezone:
+    GMT
+  xcode:
+    version: 8.0
+  environment:
+    XCODE_SCHEME: nupic
+    XCODE_WORKSPACE: nupic
+    ARCHFLAGS: "-arch x86_64"
+    PYTHONPATH: "$HOME/Library/Python/2.7/lib/python/site-packages"
+    PATH: "$HOME/Library/Python/2.7/bin:$PATH"
+    PYBIN: "$HOME/Library/Python/2.7/bin"
+  pre:
+    - pip install --user --ignore-installed --upgrade --verbose virtualenv
+    - ln -s ~/Library/Python/2.7/bin/virtualenv /usr/local/bin/virtualenv
+
+dependencies:
+  override:
+    - curl --silent --show-error --retry 5 -O http://releases.numenta.org/pip/1ebd3cb7a5a3073058d0c9552ab074bd/get-pip.py
+    - python get-pip.py --user --upgrade --ignore-installed pip setuptools wheel
+    # Still don't get the user-installed version of pip, but this seems manageable.
+    - pip --version
+    - pip install --use-wheel --user -r bindings/py/requirements.txt --quiet || exit
```

Comment:
In a future update, consider collapsing this and the following two pip lines into a single line and moving the contents of the `override` section into a script at `ci/circleci` that is invoked here for consistency with current CI integrations.